packages/assets-controllers/src/MultichainBalancesController/MultichainBalancesController.test.ts

```diff
@@ -419,25 +426,250 @@ describe('BalancesController', () => {
     expect(controller.state.balances[mockBtcAccount.id]).toStrictEqual({});
   });
 
-  it('updates balances when receiving "MultichainAssetsController:stateChange" event', async () => {
-    const { controller, messenger } = setupController();
+  describe('when MultichainAssetsController:newAccountAssets is fired', () => {
+    it('updates balances when receiving "MultichainAssetsController:newAccountAssets" event and state is empty', async () => {
+      const mockListSolanaAccounts = [
+        {
+          address: 'EBBYfhQzVzurZiweJ2keeBWpgGLs1cbWYcz28gjGgi5x',
+          id: uuidv4(),
+          metadata: {
+            name: 'Solana Account 1',
+            importTime: Date.now(),
+            keyring: {
+              type: KeyringTypes.snap,
+            },
+            snap: {
+              id: 'mock-sol-snap',
+              name: 'mock-sol-snap',
+              enabled: true,
+            },
+            lastSelected: 0,
+          },
+          scopes: [SolScope.Devnet],
+          options: {},
+          methods: [SolMethod.SendAndConfirmTransaction],
+          type: SolAccountType.DataAccount,
+        },
+        {
+          address: 'GMTYfhQzVzurZiweJ2keeBWpgGLs1cbWYcz28gjGgi5x',
+          id: uuidv4(),
+          metadata: {
+            name: 'Solana Account 2',
+            importTime: Date.now(),
+            keyring: {
+              type: KeyringTypes.snap,
+            },
+            snap: {
+              id: 'mock-sol-snap',
+              name: 'mock-sol-snap',
+              enabled: true,
+            },
+            lastSelected: 0,
+          },
+          scopes: [SolScope.Devnet],
+          options: {},
+          methods: [SolMethod.SendAndConfirmTransaction],
+          type: SolAccountType.DataAccount,
+        },
+      ];
```

Comment:
Let's move this outside of the `it` call, so we can re-use it for the other tests

Reply:
[ace5334](https://github.com/MetaMask/core/pull/5761/commits/ace5334a446a025b1227e2ba55c90471edc96d73)